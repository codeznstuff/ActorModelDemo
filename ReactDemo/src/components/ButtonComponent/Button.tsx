import React, { FC, useLayoutEffect, useState } from 'react';
import { buttonProps } from '../../@types/constants';
import styles from './Button.module.scss';
import { useStateContext } from '../../context/context';

export const Button: FC<buttonProps> = (props: buttonProps) => {
  const { currentKeyPress } = useStateContext();
  const [buttonEffect, setButtonEffect] = useState<boolean>(false);

  /* Ignore in test coverage report due to simplicity*/
  /* istanbul ignore next */
  useLayoutEffect(() => {
    if (currentKeyPress === props.id) {
      setButtonEffect(true);
    } else {
      setButtonEffect(false);
    }
  }, [currentKeyPress, props.id]);

  const getClassName = (operator?: boolean) => {
    let cls = styles.btn_wrapper + ' ';
    if (operator) {
      cls += buttonEffect ? styles.operator_hover : styles.operator;
    } else {
      cls += buttonEffect ? styles.btn_wrapper_hover : '';
    }
    return cls;
  };
  return (
    <div id={'Button' + props.id} className={getClassName(props.operator)} onClick={() => props.handleClick(props.value)} data-testid={props.id}>
      {props.value}
    </div>
  );
};

export default Button;
