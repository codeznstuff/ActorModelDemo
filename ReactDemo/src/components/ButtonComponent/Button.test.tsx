import React from 'react';
import { cleanup } from '@testing-library/react'
import { Button } from './Button';

import Enzyme, { shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe("ButtonComponent", () => {
    let button: ShallowWrapper;
    let mockClick: jest.Mock;
    beforeEach(() => {
        mockClick = jest.fn();
        button = shallow(<Button id="5" value="5" handleClick={mockClick} />)
    });
    it('Button Snapshot should match', () => {
        expect(button.debug()).toMatchSnapshot();
    })
    it('Button should have text equal to the value', () => {
        expect(button.text()).toBe("5");
    });
    it('Should call function on button click', async () => {
        button.simulate('click');
        expect(mockClick).toBeCalled();
    });

    afterEach(cleanup);
})

