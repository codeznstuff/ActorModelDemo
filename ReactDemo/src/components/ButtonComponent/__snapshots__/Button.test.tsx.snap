// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent Button Snapshot should match 1`] = `
"<div id=\\"Button5\\" className=\\"btn_wrapper \\" onClick={[Function: onClick]} data-testid=\\"5\\">
  5
</div>"
`;
